### PR TITLE
feat: add CAST function placeholder.

### PR DESCRIPTION
**Pull request by WyattJia**
Signed-off-by: Wyatt Jia <i@eventloop.live>

**What this PR does**:

Add `CAST` function placeholder logic for whole `CAST` entrypoints.

**Which issue(s) this PR fixes**:

ref #465 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/runtime/function2/cast.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+package function2
+
+import (
+	"context"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/proto"
+	"github.com/pkg/errors"
```
Comment (dongzl) [inline pkg/runtime/function2/cast.go:26]: format import, use this tool:

```
go get -u github.com/dubbogo/tools/cmd/imports-formatter
imports-formatter
```

File: pkg/runtime/function2/cast_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+package function2
+
+import (
+	"context"
+	"testing"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/proto"
+	"github.com/stretchr/testify/assert"
```
Comment (dongzl) [inline pkg/runtime/function2/cast_test.go:27]: The same.

**Review by dongzl — CHANGES_REQUESTED** [review]

File: pkg/runtime/function2/cast.go (hunk 1 above)
Comment (WyattJia) [reply pkg/runtime/function2/cast.go:27]: Formatted and thanks for this tool.

File: pkg/runtime/function2/cast.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+func (c castFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	val, err := inputs[0].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	signCast := func(i uint) *int {
+		var sign int
+		return &sign
+	}
```
Comment (jjeffcaii) [inline pkg/runtime/function2/cast.go:49]: Too simple, and no need to return `*int`. 

Please consider these cases:

1. CAST(3.14 AS SIGNED)
2. CAST(3.14 AS UNSIGNED)
3. CAST(3.14 AS CHAR)
4. CAST(3.14 AS NCHAR)
5. ... (TBD: mysql8.0+)

So I think it should accept 2 arguments, first is the input value to be cast, second is the target type in string, for example:
```go
myCastFunc.Apply(ctx, 3.14, "CHAR")
```

PS: `CONVERT` has similar behavior.

File: pkg/runtime/function2/cast.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+	switch v := val.(type) {
+	case uint8, uint16, uint32, uint64:
+		return v, nil
+	case uint:
+		return signCast(v), nil
+	default:
+		return v, nil
+	}
+}
```
Comment (jjeffcaii) [inline pkg/runtime/function2/cast.go:59]: Support more input value type, for example: `CAST('3.14' AS SIGNED)`

**Review by jjeffcaii — CHANGES_REQUESTED** [review]
Pls fix

File: pkg/runtime/function2/cast.go (hunk 3 above)
Comment (WyattJia) [reply pkg/runtime/function2/cast.go:49]: > Too simple, and no need to return `*int`. 
> 
> 
> 
> Please consider these cases:
> 
> 
> 
> 1. CAST(3.14 AS SIGNED)
> 
> 2. CAST(3.14 AS UNSIGNED)
> 
> 3. CAST(3.14 AS CHAR)
> 
> 4. CAST(3.14 AS NCHAR)
> 
> 5. ... (TBD: mysql8.0+)
> 
> 
> 
> So I think it should accept 2 arguments, first is the input value to be cast, second is the target type in string, for example:
> 
> ```go
> 
> myCastFunc.Apply(ctx, 3.14, "CHAR")
> 
> ```
> 
> 
> 
> PS: `CONVERT` has similar behavior.

Sure I will fix those. But this pull request just for `CAST` function placeholder,not a complete implementation, for other `CAST` function developers to use this file for entrypoint.

**Review by jjeffcaii — APPROVED** [review]

**Review by dongzl — APPROVED** [review]

**Review by Mulavar — APPROVED** [review]

**Pull request merged**